File: Basics/OOPS/JAVA/CreditCard.java
public class CreditCard extends Card{
    public CreditCard(String cardNumber, String cardHolderName) {
        super(cardNumber, cardHolderName);
    }

    @Override
    public void pay(double amount) {
        System.out.println("Paid " + amount + " using Credit Card: " + getCardHolderName() + " with card number " + getCardNumber());
    }
}

File: Basics/OOPS/CPP/CreditCard.cpp
#include<iostream>
#include "Card.cpp"
using namespace std;

class CreditCard : public Card{
     public: 
     CreditCard(const int &number, const string &name) : Card(number, name){}
     
     void processPayment(double amount) override{
            cout << "Processing credit card payment of $" << amount << " for card holder: " << getCardHolderName() << endl;
     }
};